core/embed/lib/error_handling.h

```diff
@@ -20,6 +20,167 @@
 #ifndef LIB_ERROR_HANDLING_H
 #define LIB_ERROR_HANDLING_H
 
+// Suppresses the intellisense error in VSCode
+#ifndef __FILE_NAME__
+#define __FILE_NAME__ __FILE__
+#endif
+
+// Status code type
+typedef struct {
+  // Do not access this field directly,
+  // use `ts_ok()` and `ts_error()` macros.
+  uint32_t code;
+} ts_t;
+
+// Build status code from any 16-bit value.
+//
+// Status codes are hardened against fault injections
+// by storing the same value in the upper 16 bits.
+#define TS_BUILD(code) ((ts_t){(code) | ((code) << 16)})
```

Comment:
Is there any reason not to use `(const ts_t)`?

Reply:
Probably not, but I'm not sure if adding `const` here brings any advantage.

Reply:
As a rule of thumb if it can be const, I make it const. It generally helps to detect bugs early, e.g. something like `TS_OK = verify_status()`.